### PR TITLE
Add edit description

### PR DESCRIPTION
**Pull request by kgudel**
Address #717 

This PR is the first piece of #717 it adds a description field to each edit which is persisted through when an edit fails and is passed through the `ValidationErrorConverter` to be returned through the API.

### CODE REVIEW

**Comment by jmarin** [comment]
I've only taken a quick look at this, but I think this approach could be improved. Ticket #717 is asking for additional metadata about the edits. This is all about providing the user with additional, human readable descriptions of the edits (as per https://github.com/cfpb/hmda-platform-ui/issues/96). I don't think this needs to be baked into the edit model, but rather be a lookup that gets included into the `json` response. At the persistence and validation level, we don't care about these messages. It's only on the presentation layer (api project) where we need to incorporate these. On ticket #724 the additional fields and content are described in a file. A possible approach might be to load this when necessary, and link to the validation errors by edit id. I don't think we want to persist all these fields, since they will most likely change (and it's extra stuff in the journal that we don't need)

**Comment by kgudel** [comment]
@jmarin Those are really good points. I hadn't considered the burden of persisting all of that extra information. I have refactored it to get the descriptions from a flat file which has the added benefit of already having all of the edit descriptions in it preventing the need for a task to add descriptions to each edit individually.

File: Documents/API.md [hunk 1]
```diff
@@ -275,7 +277,8 @@ All endpoints in the `/institutions` namespace require two headers (see "Authori
         {
             "edits": [
                 {
-                   "edit": "q007",
+                   "edit": "Q007",
+                   "description: "Description of Q007",
```
Comment (schbetsy) [inline Documents/API.md]: This `"description` and two others below are missing closing quotes, which is why the highlighting here is weird.

File: validation/src/main/scala/hmda/validation/engine/ValidationError.scala [hunk 2]
```diff
@@ -7,7 +7,8 @@ case object Quality extends ValidationErrorType
 case object Macro extends ValidationErrorType
 
 // errorID = Loan ID (LAR) or Agency Code + Respondent ID (TS)
-case class ValidationError(errorId: String, name: String, errorType: ValidationErrorType)
+case class ValidationError(errorId: String, metaData: ValidationErrorMetaData, errorType: ValidationErrorType)
+case class ValidationErrorMetaData(name: String)
```
Comment (schbetsy) [inline validation/src/main/scala/hmda/validation/engine/ValidationError.scala]: I don't understand yet what the `ValidationErrorMetaData` class is adding. Will it eventually hold more metadata than just the edit name?

File: model/jvm/src/main/scala/hmda/model/edits/EditMetaDataLookup.scala [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+      )
+
+    }.toSeq
+  }
+}
+
+case class EditMetaData(
+  category: String,
+  editType: String,
+  fieldNames: String,
```
Comment (schbetsy) [inline model/jvm/src/main/scala/hmda/model/edits/EditMetaDataLookup.scala]: I noticed in the file that `fieldNames` only lists one field for each edit, though most edits have several relevant fields. (for example, `V275` depends on `applicationDate` and `actionTakenDate`, but only "Action Taken Date" is listed in the file.)

This isn't relevant for this PR, but it seems like it will become relevant soon.

File: api/src/main/scala/hmda/api/http/ValidationErrorConverter.scala [hunk 4]
```diff
@@ -1,35 +1,45 @@
...
+import hmda.model.edits.{ EditMetaData, EditMetaDataLookup }
 import hmda.validation.engine._
 
 trait ValidationErrorConverter {
 
+  val editDescriptions = EditMetaDataLookup.values
+
   def validationErrorsToEditResults(tsErrors: Seq[ValidationError], larErrors: Seq[ValidationError], validationErrorType: ValidationErrorType) = {
+
+    def findEditDescription(editName: String): String = {
```
Comment (kgudel) [inline api/src/main/scala/hmda/api/http/ValidationErrorConverter.scala]: Absolutely, this also opens up the ability to individually test this function.

File: validation/src/main/scala/hmda/validation/engine/ValidationError.scala (hunk 2 above)
Comment (schbetsy) [inline validation/src/main/scala/hmda/validation/engine/ValidationError.scala]: Would it be possible (without a large time commitment) to remove `ValidationErrorMetaData` from this PR? Since that class won't be used until a future PR, I think it doesn't make sense to code review and merge it now.

I don't want to waste your time doing git stuff, but I also want to wait and evaluate `ValidationErrorMetaData` in context, since it's a big change in a confusing part of the code.

File: api/src/test/scala/hmda/api/model/ModelGenerators.scala [hunk 5]
```diff
@@ -12,7 +12,7 @@ trait ModelGenerators {
     for {
       status <- Gen.oneOf("OK", "SERVICE_UNAVAILABLE")
       service = "hmda-api"
-      time = Calendar.getInstance().getTime().toString
+      time = Calendar.getInstance().getTime.toString
```
Comment (jmarin) [inline api/src/test/scala/hmda/api/model/ModelGenerators.scala]: Why this change? I find that with `Java` getters it's easier to read if we keep the parens as in the original language / api. Either that, or also change `getInstance()` to `getInstance`

File: validation/src/main/scala/hmda/validation/rules/lar/quality/Q068.scala [hunk 6]
```diff
@@ -19,4 +19,5 @@ object Q068 extends EditCheck[LoanApplicationRegister] with ApplicantUtils {
         (app.coSex not equalTo(4))
     }
   }
+
```
Comment (jmarin) [inline validation/src/main/scala/hmda/validation/rules/lar/quality/Q068.scala]: Same comment on these lines, seem to unnecessarily add a blank space to existing code, without changing behavior

**Comment by schbetsy** [comment]
Any more comments on this? @jmarin ?

**Comment by jmarin** [comment]
No. I'm good
